kubernetes/core/src/main/kotlin/com/google/container/tools/core/settings/KubernetesSettingsConfigurable.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.core.settings
+
+import com.google.common.annotations.VisibleForTesting
+import com.google.kubernetes.tools.core.util.CoreBundle
+import com.intellij.openapi.fileChooser.FileChooserDescriptor
+import com.intellij.openapi.options.Configurable
+import com.intellij.openapi.ui.TextFieldWithBrowseButton
+import com.intellij.ui.border.IdeaTitledBorder
+import com.intellij.ui.layout.panel
+import java.awt.Insets
+import javax.swing.JComponent
+
+/**
+ * Creates a "Kubernetes" menu item under the "Google" menu item in the IDE Settings.
+ */
+class KubernetesSettingsConfigurable : Configurable {
+
+    @VisibleForTesting
+    val skaffoldBrowser = TextFieldWithBrowseButton()
+
+    init {
+        skaffoldBrowser.addBrowseFolderListener(
+                CoreBundle.message("kubernetes.settings.dependencies.skaffold.browse.title"),
+                null /*description*/,
+                null /*project*/,
+                FileChooserDescriptor(
+                        true /*chooseFiles*/,
+                        false /*chooseFolders*/,
```

Comment:
Will we choose files with this file chooser? Would it make sense to actually select folder where skaffold executable is located?

Reply:
I was thinking that this will choose the actual executable file.

In linux, for example, its placed (according to the install instructions) in `/usr/local/bin`. I think it makes more sense to browse to the actually skaffold executable rather than this bin directory.

wdty?

Reply:
Also make note of the wording of the label on the browser control `Path to Skaffold executable` in case you think it could be more clear.

Reply:
Nothing wrong with selecting a particular file in general, but it seems more of a convention to select a directory where some tool lives - i.e. Maven, Gradle, our selection for custom Cloud SDK, all of this in the same IntelliJ. It could be more in line with IDE experience to select a directory where something is, or you think it might be easier to actually point to skaffold file?

Reply:
The directory selection seems more appropriate for an sdk style tool where, for you example, you have a bin directory with multiple executables, whereas skaffold is just a single binary.

One example like this that I could find is the docker tool selection (Build, Execution, Deployment -> Docker > Tools) where you browse to the docker machine and docker compose executable.

Reply:
good point with SDK style directories. Let's use file chooser, hopefully we won't have to worry much about this anyway once managed CLIs are with us.